app/lib/applets/substitutions/parser.dart

```diff
@@ -193,4 +198,59 @@ class SubstitutionsParser extends AppletParser<SubstitutionPlan> {
     int second = int.parse(match.group(6) ?? "00");
     return DateTime(year, month, day, hour, minute, second);
   }
-}
\ No newline at end of file
+
+  String parseStunde(String stunde) {
+    final numbers =
+        RegExp(r'\d+').allMatches(stunde).map((m) => m.group(0)!).toList();
+    if (numbers.isEmpty || numbers.length > 2) return stunde;
+    return numbers.length == 2 ? '${numbers[0]} - ${numbers[1]}' : numbers[0];
+  }
+
+  List<SubstitutionInfo> parseInformationTables(
```

Comment:
This does not seem to work for my testing schools. I will investigate later and maybe adjust the parsing.